### ISSUE TITLE
The focus management for wizards steps is not logical

### ISSUE THREAD

**Issue by chlane** [vmware/clarity#4834]
Component: Wizards
WCAG_Success_Criterion: 2.4.3 Focus Order
WCAG_Conformance_Level: A
Test URL:  https://clarity.design/documentation/wizards#examples

Description: This is a Clarity issue. Wizards actually have 2 titles. When they initially open, focus should go to the first heading at the top left corner of the wizard dialog. Then, when the user moves to subsequent steps in the wizard, focus should to the second heading at the top of the right side of the wizard.

See the relevant HTML below.

```
<h1 class="clr-wizard-title"><clr-wizard-title>Medium-Sized Wizard</clr-wizard-title></h1>

<h2 class="modal-title ng-tns-c167-4"><span tabindex="-1" class="modal-title-text">Page 1<!----></span><!----></h2>
```

Steps to Reproduce:
- Go to the Test URL 
- Open the medium modal (or any other) and observe that focus does not move to the h1 at the top left
- Advance through the wizard and observe that focus does not move to the h2 at the top of the right side of the wizard

User Impact:
Sighted keyboard-only and blind screen reader users will not have a logical tabbing or reading experience

Recommended Remediation:
Use tabindex="-1" on the headings and use focus() to focus them as needed.

**Comment by chlane** [comment]
As pointed out by @gkhadK the h1 should be the previous sibling of the nav element as seen below.
```
<h1 class="clr-wizard-title"><clr-wizard-title>Medium-Sized Wizard</clr-wizard-title></h1>
<nav class="modal-nav clr-wizard-stepnav-wrapper ng-tns-c167-4">
```
This will be less confusing for screen reader users because the current implementation changes the structure from screen reader perspective and creates confusion. Just imagine screen reader reading it as a navigation landmark and a h1.

**Comment by gkhadK** [comment]
In addition to that focus order should be logical and intuitive so users know what's coming next or where they are currently in the process (for example: using step indicators in the footer).

**Comment by gnomeontherun** [comment]
Hi there, this is an automated message.

In an effort to dedupe and track accessibility reports consistently, this issue is being closed and tracked internally with our Accessibility team. It will still be triaged and addressed as part of ongoing work.
